src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -197,20 +197,28 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
             final boolean isSuccessful = done.await(settings.getKafkaSendTimeoutMs(), TimeUnit.MILLISECONDS);
 
             if (!isSuccessful) {
+                failBatch(batch, "timed out");
                 throw new EventPublishingException("Timeout publishing events");
             }
         } catch (InterruptedException e) {
+            failBatch(batch, "internal error");
             throw new EventPublishingException("Error publishing message to kafka", e);
         }
     }
 
+    private void failBatch(List<BatchItem> batch, final String reason) {
+        for (final BatchItem item : batch) {
+            item.updateStatusAndDetail(EventPublishingStatus.FAILED, reason);
+        }
+    }
+
     private Callback kafkaSendCallback(final BatchItem item, final CountDownLatch done) {
         return (metadata, exception) -> {
             if (exception == null) {
-                item.setPublishingStatus(EventPublishingStatus.SUBMITTED);
+                item.updateStatusAndDetail(EventPublishingStatus.SUBMITTED, "");
             } else {
                 LOG.error("Failed to publish event " + item.getEvent().toString(), exception);
```

Comment:
I think we should not log event here, but not because of security resons but because it's not really important what is the content of message here. I believe a failure with publishing to kafka doesn't depend on message content, because kafka doesn't care what is inside a message.